### DISCUSSION TITLE
makeAutoObservable and make

### DISCUSSION THREAD

**Discussion by MutelyMilo** [mobxjs/mobx#2847] [Q&A] [answered]
Hello, dear author:

I did not find a description of the second and third arguments in the method...

Demo:
`makeAutoObservable(this, { rootStore: false });`

rootStore: false

What does that mean?

In mobx6, the documentation feels like it doesn't go deep enough, and I'm afraid of missing the meaning of some features and parameters.

**Comment by kubk** [comment, marked as answer]
It's documented here: https://mobx.js.org/observable-state.html#makeautoobservable

"Members marked with false in the overrides argument will not be annotated"

Function `makeAutoObservable` automatically converts properties to observables, methods to actions and getters to computed. Using `false` in second arguments allows skipping this conversion for a specific field.

**Reply by MutelyMilo** [reply]
That's pretty straightforward about my problem, Thanks

**Comment by urugator** [comment]
2nd argument are [annotations](https://mobx.js.org/observable-state.html#available-annotations) - same as `makeObservable`, but are optional - it allows you to override auto behavior for specific field. `false` means ignore, `true` means auto (has the same effect as not specifying override)

3rd argument are [options](https://mobx.js.org/observable-state.html#options-)

**Reply by MutelyMilo** [reply]
Thank you very much. It was an oversight on my part

**Comment by mattleonowicz** [comment]
@kubk @urugator I have a followup question to this. This rootStore is always marked with `false` throughout the docs. Why is that? Is it to avoid double-observing it, because it's already observed? I don't have any issues when I forget to to this.

**Reply by urugator** [reply]
`rootStore` field should be `readonly`, so no need to make it observable, but also no harm in doing so - field becomes observable and value is ignored, because it's instance of a class.

**Reply by kubk** [reply]
@mattleonowicz Good catch, I believe the documentation should be updated: https://github.com/mobxjs/mobx/pull/3775

As for me, I never exclude class instances from `makeAutoObservable`.
